wordpress-mobile/WordPress-iOS
# MediaLibrary: directory types

This first PR is the base for the subsequent Media exporter PRs, which will be merged here before merging into `develop`.

In this PR we're introducing the idea of using specific types of local Media directories. Between `.uploads`, `.cache`, and `.temporary` we're able to control where/how Media files are stored on disk and the inherent handling of the files based on iOS.

We'll be using `.uploads` to hang on to Media files that need to be uploaded. The `.cache` is planned to be used for downloading existing Media files that can be discarded by iOS or the user. And lastly, `.temporary` is handy for not worrying too much about cleaning up anything written to disk, as seen with unit tests for example.

Currently the default is `.uploads` which uses the same directory path we've always used.

Note: I included a goofy default `func` as `localDirectory()` and `makeLocalMediaURL()` to satiate Obj-C.

To test:
1. Build, make sure everything builds.
2. Run all unit tests, make sure everything passes.
3. Test that inserting and posting an image from the device library picker works.
4. Test that inserting and posting an image from the Media library picker works.
5. Test that the internal Media Library collection loads as expected.

Needs review: @frosty and @SergioEstevao ready for round one 😅
